yt_dlp/extractor/netverse.py

```diff
@@ -0,0 +1,174 @@
+from .common import InfoExtractor
+from ..utils import (
+    traverse_obj,
+    ExtractorError,
+    # GeoRestrictedError,
+)
+
+
+class NetverseBaseIE(InfoExtractor):
+    def get_required_json(self, url, data=None):
+        match = self._match_valid_url(url).groupdict()
+        display_id, sites_type = match['display_id'], match['type']
+
+        if sites_type == 'watch' or sites_type == 'video':
+            media_api_url = f'https://api.netverse.id/medias/api/v2/watchvideo/{display_id}'
+        elif sites_type == 'webseries':
+            media_api_url = f'https://api.netverse.id/medias/api/v2/webseries/{display_id}'
+
+        json_data = self._download_json(media_api_url, display_id, data=data)
```

Comment:
```suggestion
        json_data = self._download_json(
            f'https://api.netverse.id/medias/api/v2/{self._ENDPOINTS[sites_type]}/{display_id}',
            display_id, data=data)
```
and define this above
```py
    _ENDPOINTS = {
       'watch': 'watchvideo',
       ...
    }
```